extract data and follow this:
- which problem you are trying to solve, identify objective
- start performing linear regresison (pick one x and y)
- explain and narrate concise description
- do the correlation, summary, explain the constant b and m
- find out how they are corerlated and see of my linear regression cna work that well.
- no discrete only continous
- using ols method and send as early as you can.


- san diego miramar college library
- go to ACM and search linear research paper,pick one and write notes on the paper ( have a feel of how it look like to publish a paper)


# ***Linear Regression Data Analysis***

In [42]:
import statsmodels.api as sm
import pandas as pd
import plotly.express as px
import plotly.io as pio
import numpy as np 
pio.renderers.default = "iframe"
import requests
from io import StringIO
url = "https://www.openintro.org/data/csv/fastfood.csv"
headers = {'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.3'}
response = requests.get(url, headers=headers)
response.raise_for_status() # Raise an exception for bad status codes
df = pd.read_csv(StringIO(response.text))

In [43]:
df.head()

,restaurant,item,calories,cal_fat,total_fat,sat_fat,trans_fat,cholesterol,sodium,total_carb,fiber,sugar,protein,vit_a,vit_c,calcium,salad
0,Mcdonalds,Artisan Grilled Chicken Sandwich,380,60,7,2.0,0.0,95,1110,44,3.0,11,37.0,4.0,20.0,20.0,Other
1,Mcdonalds,Single Bacon Smokehouse Burger,840,410,45,17.0,1.5,130,1580,62,2.0,18,46.0,6.0,20.0,20.0,Other
2,Mcdonalds,Double Bacon Smokehouse Burger,1130,600,67,27.0,3.0,220,1920,63,3.0,18,70.0,10.0,20.0,50.0,Other
3,Mcdonalds,Grilled Bacon Smokehouse Chicken Sandwich,750,280,31,10.0,0.5,155,1940,62,2.0,18,55.0,6.0,25.0,20.0,Other
4,Mcdonalds,Crispy Bacon Smokehouse Chicken Sandwich,920,410,45,12.0,0.5,120,1980,81,4.0,18,46.0,6.0,20.0,20.0,Other


## identify the relationship between calorie and cholesterol in fast food, will cholesterol also increase for food with more calorie.

first we check whether the data sets has any NaN values.

In [44]:
print(df.isna().sum())

print(df.info())

restaurant       0
item             0
calories         0
cal_fat          0
total_fat        0
sat_fat          0
trans_fat        0
cholesterol      0
sodium           0
total_carb       0
fiber           12
sugar            0
protein          1
vit_a          214
vit_c          210
calcium        210
salad            0
dtype: int64
<class 'pandas.DataFrame'>
RangeIndex: 515 entries, 0 to 514
Data columns (total 17 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   restaurant   515 non-null    str    
 1   item         515 non-null    str    
 2   calories     515 non-null    int64  
 3   cal_fat      515 non-null    int64  
 4   total_fat    515 non-null    int64  
 5   sat_fat      515 non-null    float64
 6   trans_fat    515 non-null    float64
 7   cholesterol  515 non-null    int64  
 8   sodium       515 non-null    int64  
 9   total_carb   515 non-null    int64  
 10  fiber        503 non-null    float64
 11  sugar        515 no

vitamin A, C and calcium is missing a lot of values so we will not be assesing those.

In [45]:
fig = px.scatter(df, x = "calories", y = "cholesterol", title = "Calorie vs cholesterol",
                labels={"calories": "Calories (kcal)", "cholesterol": "Cholesterol (mg)"},
                color = "restaurant")
fig.show()

we can see a faintly positively correlated relationship between calories and cholesterol. proving that food with high calorie in fast food restaurant often asscociated with high cholesterol.

in order to fully see the relationship between the calorie and the cholesterol, we can use the linear regression line to map out a linear relationship between the two variable. 

we will use the function polyfit to get the variable for the slope and the y intercept of the line. 

In [46]:
x = df["calories"]
y = df["cholesterol"]
m , b = np.polyfit(x,y,1)

print("slope:", m)
print("y-intercept:", b)


df["y_pred_cholesterol"] = m * df["calories"] + b

fig = px.scatter(df, x = "calories", y = "cholesterol",
                 labels={"calories": "Calories (kcal)", "sodium": "Sodium (mg)"},
                 color = "restaurant")

fig.add_scatter( x = df["calories"], y = df["y_pred_cholesterol"], name = "Regression prediction line", line=dict(color="red"))

fig.show()



slope: 0.17050674400670005
y-intercept: -18.067871740411597


## we can see the line following the trend of each the point, mapping the relationship between calorie and cholesterol in fast food Linearly

the y intercept value b is the the mg of cholesterol if the calorie in the food were to be zero. However, the value of B is negative, therefore we can say that a fastfood item can have some calorie and zero cholesterol and not the other way around.

the slope tells how much the cholesterol is with respect to calorie, in which this value is 0.17. therefore, for every 1 calorie being added to the food, about 0.17 mg of cholesterol is also added.

# OSL METHOD

In [47]:
X = sm.add_constant(df["calories"])  # adds intercept
model = sm.OLS(df["cholesterol"], X).fit()
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:            cholesterol   R-squared:                       0.581
Model:                            OLS   Adj. R-squared:                  0.581
Method:                 Least Squares   F-statistic:                     712.4
Date:                Wed, 15 Jul 2026   Prob (F-statistic):           4.68e-99
Time:                        17:18:49   Log-Likelihood:                -2641.1
No. Observations:                 515   AIC:                             5286.
Df Residuals:                     513   BIC:                             5295.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const        -18.0679      3.841     -4.704      0.0

- we can see the coef value as the y intercept and the 0.1705 as the slope

- the "std err" measures how much the values "wooble" around the line, meaning that ,most of the true value is + - 3.841 in the y direction and 0.006 in the x direction.

- this means thta when we use the regression line to estimate a value for cholesterol based on the calorie in the fast food, we would get a value  y = + - 3.84mg

equation for the linear regression line value estimate is:

y = 0.117x - 18.06 +/- 3.84 